### DISCUSSION TITLE
extracting paragraphs from pdf

### DISCUSSION THREAD

**Discussion by vasnt** [phiresky/ripgrep-all#171] [Q&A]
by multiple keywords (from a given list) search , Is it possible to extract (copy) matching paragraphs & lines from pdf/s  & store them in any text/markdown file? How? along with page/file name reference.

**Comment by phiresky** [comment]
your question is kinda vague and the answer really depends on what exactly you want and how your document is formatted. if you can formulate your match criteria as a regex you can extract the content you want. chatgpt could help you formulate a regex. rg doesn't have an inherent model of paragraphs, lines should be easy though.

if you want specific formatting of the output you'll probably have to use some other postprocessing tool (possibly using rga with --json output)
